This patch is necessary to use real transparency with tint2.
It can also be useful for any application which needs to know
if there is a compositing manager running.
See http://bugs.freedesktop.org/show_bug.cgi?id=26090


diff -Nur xcompmgr-dana/xcompmgr.c xcompmgr-dana.new/xcompmgr.c
--- xcompmgr-dana/xcompmgr.c	2010-01-25 21:53:13.000000000 +0100
+++ xcompmgr-dana.new/xcompmgr.c	2010-06-05 09:42:38.000000000 +0200
@@ -1991,6 +1991,21 @@
     free(buf);
 
     XSetSelectionOwner (dpy, a, w, 0);
+    
+    /* Send a ClientMessage notification so other apps
+       can detect the compositing manager started */
+    XClientMessageEvent ev;
+    ev.type = ClientMessage;
+    ev.window = RootWindow (dpy, scr);
+    ev.message_type = a;
+    ev.format = 32;
+    ev.data.l[0] = CurrentTime;
+    ev.data.l[1] = a;
+    ev.data.l[2] = w;
+    ev.data.l[3] = 0;
+    ev.data.l[4] = 0;
+    XSendEvent(dpy, RootWindow (dpy, scr), False, StructureNotifyMask, (XEvent*)&ev);
+
 }
 
 int
